packages/dev/core/src/Collisions/pickingInfo.ts

```diff
@@ -72,20 +72,30 @@ export class PickingInfo {
             return null;
         }
 
-        const indices = this.pickedMesh.getIndices();
+        let indices = this.pickedMesh.getIndices();
 
-        if (!indices) {
-            return null;
+        if (indices?.length === 0) {
+            indices = null;
         }
 
         let result: Vector3;
 
+        const tmp0 = TmpVectors.Vector3[0];
+        const tmp1 = TmpVectors.Vector3[1];
+        const tmp2 = TmpVectors.Vector3[2];
+
         if (useVerticesNormals) {
             const normals = <FloatArray>this.pickedMesh.getVerticesData(VertexBuffer.NormalKind);
 
-            let normal0 = Vector3.FromArray(normals, indices[this.faceId * 3] * 3);
-            let normal1 = Vector3.FromArray(normals, indices[this.faceId * 3 + 1] * 3);
-            let normal2 = Vector3.FromArray(normals, indices[this.faceId * 3 + 2] * 3);
+            let normal0 = indices

```

Comment:
We could use the tmpVectors in the indexed case too to avoid creating new vectors in FromArray?

Reply:
Providing temp vectors to users might leave to unexpected results

Reply:
Makes sense. You can use FromArrayToRef